src/types.rs

```diff
@@ -102,7 +121,11 @@ pub struct IndexerConfig {
     pub to_block: Option<u64>,
 
     /// The postgres configuration.
-    pub postgres: IndexerPostgresConfig,
+    pub postgres: Option<IndexerPostgresConfig>,
+
+    /// GCP configuration, if exists
+    #[serde(rename = "gcpBigQuery", skip_serializing_if = "Option::is_none")]
+    pub gcp_bigquery: Option<IndexerGcpBigQueryConfig>,
```

Comment:
do we throw if none of these are defined? and if both are defined we also throw?

Reply:
added a panic at the end of the factory method but lmk if there is a more optimal way to do this, thanks.